venues/aclweb.org/NAACL/2022/migrated-submissions/post_ethics_review_invitation.py

```diff
@@ -0,0 +1,102 @@
+import argparse
+from re import sub
+import openreview
+from tqdm import tqdm
+import csv
+import tracks
+import flagged_papers
+
+"""
+OPTIONAL SCRIPT ARGUMENTS
+
+    baseurl -  the URL of the OpenReview server to connect to (live site: https://openreview.net)
+    username - the email address of the logging in user
+    password - the user's password
+
+"""
+parser = argparse.ArgumentParser()
+parser.add_argument('--baseurl', help="base URL")
+parser.add_argument('--username')
+parser.add_argument('--password')
+args = parser.parse_args()
+client = openreview.Client(baseurl=args.baseurl, username=args.username, password=args.password)
+sac_name_dictionary = tracks.sac_name_dictionary
+program_chairs_id = 'aclweb.org/NAACL/2022/Conference/Program_Chairs'
+
+
+ethics_review_super = openreview.Invitation(
+    id = "aclweb.org/NAACL/2022/Conference/-/Ethics_Review",
+    readers = ["everyone"],
+    writers = ["aclweb.org/NAACL/2022/Conference"],
+    signatures = ["aclweb.org/NAACL/2022/Conference"],
+    reply = {
+        "writers":{
+            "values-copied":[
+                "aclweb.org/NAACL/2022/Conference",
+                "{signatures}"
+            ]
+        },
+        "signatures":{
+            "values-regex":"~.*"
+        },
+        "content": {
+            "recommendation": {
+                "order": 1,
+                "value-radio": [
+                    "1 - No serious ethical issues",
+                    "2 - Serious ethical issues that need to be addressed in the final version",
+                    "3 - Paper should be rejected due to ethical issues"
+                ],
+                "description": "Please select your ethical recommendation",
+                "required": True
+                },
+            "ethics_review": {
+                "order": 3,
+                "value-regex": "[\\S\\s]{1,200000}",
+                "description": "Provide justification for your suggested ethics issues. Add formatting using Markdown and formulas using LaTeX. For more information see https://openreview.net/faq.",
+                "required": True,
+                "markdown": True
+                }
+        }
+    }
+)
+client.post_invitation(ethics_review_super)
+submissions_forum_list = flagged_papers.flagged_papers
+
+blind_submission_by_number = { s.number: s for s in openreview.tools.iterget_notes(client, invitation='aclweb.org/NAACL/2022/Conference/-/Blind_Submission')}
+submission_by_number = { s.number: s for s in openreview.tools.iterget_notes(client, invitation='aclweb.org/NAACL/2022/Conference/-/Submission')}
+# For Each submission, create reviewer group, add reviewer group and AC group to readers
+for submission_number in tqdm(submissions_forum_list):
```

Comment:
are we creating Ethics_Review invitations for all papers and not only flagged ones?

Reply:
nevermind!